packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -153,7 +153,7 @@ mixin CupertinoRouteTransitionMixin<T> on PageRoute<T> {
   @override
   bool canTransitionTo(TransitionRoute<dynamic> nextRoute) {
     // Don't perform outgoing animation if the next route is a fullscreen dialog.
-    return nextRoute is CupertinoRouteTransitionMixin && !nextRoute.fullscreenDialog;
+    return !(nextRoute is CupertinoRouteTransitionMixin && nextRoute.fullscreenDialog);
```

Comment:
This is just stuck in to keep the transition playing correctly when navigating to a Material page. For the actual implementation this will either be left out or done better.